src/ui/settings/components/InstallPluginAlert.tsx

```diff
@@ -0,0 +1,64 @@
+import { clipboard } from "@metro/common";
+import { installPlugin } from "@lib/plugins";
+import { findByProps } from "@metro/filters";
+import { Forms } from "@ui/components";
+
+const { FormInput } = Forms;
+const Alert = findByProps("alertDarkStyles", "alertLightStyles").default;
+
+const Alerts = findByProps("openLazy", "close");
+
+export function showInstallPluginAlert() {
+	Alerts.openLazy({
+		importer: async function () {
+			return () => <InstallPluginAlert />;
+		}
+	});
+};
+
+const HTTP_REGEX = /^https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b(?:[-a-zA-Z0-9()@:%_\+.~#?&\/=]*)$/;
+
+export default function InstallPluginAlert() {
+	const [pluginUrl, setPluginUrl] = React.useState("");
+	const [error, setError] = React.useState("");
+
+	React.useEffect(() => {
+		clipboard.getString().then((c) => { if (HTTP_REGEX.test(c)) setPluginUrl(c) });
+	}, []);
+
+	const installPluginCallback = () => {
+		installPlugin(pluginUrl).then(() => {
+			setPluginUrl("");
+			Alerts.close();
+		}).catch((e: Error) => {
+			setError(e.message);
+		});
+	};
+
+	return (
```

Comment:
I'd prefer if this entire file was restructured into an Alerts API for plugins to use as well as Vendetta. You could have a function to display a custom modal, taking arguments for the alert props, as well as a children property, for whatever the developer may wish to embed. We could even provide pre-filled functions for common modals, like input dialogs such as this one. They would use the primary `showModal` function, with premade inputs.